lib/vast/Conversion/Common/IRsToLLVM.cpp

```diff
@@ -48,6 +49,54 @@ namespace vast::conv::irstollvm
         erase_pattern< hl::TypeDeclOp >
     >;
 
+    template< typename self_t >
+    struct value_builder
+    {
+      private:
+        const self_t &self() const { return static_cast< const self_t & >(*this); }
+
+      protected:
+
+        mlir::Value construct_value(auto &rewriter, mlir_value val) const {
+            auto op = val.getDefiningOp();
+            VAST_ASSERT(op && op->getResults().size() == 1);
+            auto trg_type = self().convert(op->getResults()[0].getType());
+            return construct_value(rewriter, op, trg_type);
+        }
+
+        mlir::Value construct_value(auto &rewriter, operation op, mlir_type type) const {
+            auto init_list = mlir::dyn_cast< hl::InitListExpr >(op);
+            if (!init_list)
```

Comment:
is only init list multi-value constructor?

Reply:
For now this is the only one we expect. If some new shows we will adjust, I am just being defensive.